### DISCUSSION TITLE
Can't get mapping Many to Many relations without join entity to work.

### DISCUSSION THREAD

**Discussion by martinkcentric** [thepirat000/Audit.NET#684] [Q&A]
Hi,

I can't get mapping Many to Many relations without join entity to work.
I followed the description [here](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md#map-many-to-many-relations-without-join-entity), and also looked at [this discussion](https://github.com/thepirat000/Audit.NET/discussions/603), but to no avail.

In my database context I have:
```
        private readonly IAuditDbContext _auditContext;

        public MyDbContext()
        {
            _auditContext = new DefaultAuditContext(this);
            _helper.SetConfig(_auditContext);
        }
```

My configuration is like this:

```
        public static void Initialize()
        {
            Audit.EntityFramework.Configuration.Setup()
                .ForAnyContext(config => config
                    .IncludeEntityObjects()
                    .IncludeIndependantAssociations());

            Audit.Core.Configuration.Setup()
                .UseEntityFramework(ef => ef
                    .AuditTypeExplicitMapper(m => m
                        .Map<Domain, Log>((evt, entry, log) =>
                        {
                            // snip
                        })
                        .Map<MappingGroup, Log>((evt, entry,Log) => 
                        { 
                            // snip
                        })
                        .MapTable<Log>("MappingGroupDomains", (entry, log) =>
                        {
                            // snip
                        })
                        .AuditEntityAction<Log>((evt, entry, log) =>
                        {
                            // snip
                        }))
                .IgnoreMatchedProperties(true));
        }
```
The lambda expression in the MapTable is never executed.
If I insert an association *and* change a 'regular' (e.g. string) property in MappingGroup,  the lambda in Map<MappingGroup, Log> is executed and I can see in the insert in `evt.EntityFrameworkEvent.Associations`.
If I insert an association without changing a regular property the lambda is not executed.
I'm using Audit.EntityFramework 26.0.0 with EntityFramework 6.5.1 - database first.

What am I doing wrong?
Thanks in Advance.

**Comment by thepirat000** [comment]
Can you share your EF entity mapping configuration?

**Comment by martinkcentric** [comment]
That's a bit sensitive, so I've removed all tables but the ones involved here.
[MyDb.zip](https://github.com/user-attachments/files/16402645/MyDb.zip)

**Reply by martinkcentric** [reply]
It doesn't work yet, but I've made some progress.

There were no added or deleted associations in the object state entries.

After adding this:

```
            var objectContext = ((System.Data.Entity.Infrastructure.IObjectContextAdapter)this).ObjectContext;
            objectContext.DetectChanges();
```

before the call to `_helper.SaveChanges`, added/deleted associations start showing up in the object state entries.

However there is some obstruction further down the path, because the lambda here:

```
                        .MapTable<Log>("MappingGroupDomains", (entry, log) =>
                        {
                            // snip
                        })
```

still doesn't get called.
